spark3/src/main/java/org/apache/iceberg/actions/Spark3CreateAction.java

```diff
@@ -187,4 +189,16 @@ protected String getMetadataLocation(Table table) {
   protected abstract Map<String, String> targetTableProps();
 
   protected abstract TableCatalog checkSourceCatalog(CatalogPlugin catalog);
+
```

Comment:
Same comment as for `BaseSparkAction` above.

For my own benefit why do the spark3 actions have a different base and is this part of  what is being addressed by Antons refactor? Not a comment on this PR specifically, just me trying to learn the history.

Reply:
Yeah, I'll fix the hierarchy. It does not make sense right now.

Reply:
The difference in Catalog Apis was a big part, Spark 2 doesn't have access to CatalogPlugin for example